mlflow/pyfunc/model.py

```diff
@@ -170,6 +183,13 @@ def __init__(self, func, signature=None):
     def _get_type_hints(self):
         return _extract_type_hints(self.func, input_arg_index=0)
 
+    def __getattribute__(self, name):
```

Comment:
Shall we make this thread safe?

Reply:
What's the cases this could be invoked by multiple threads? I think this function is called when you create an instance 🤔

Reply:
> What's the cases this could be invoked by multiple threads?

Evaluation, model serving, and potentially more depending on how users run model🙂

> I think this function is called when you create an instance 🤔

This is called when users call `model.predict` right?

Reply:
Ah yes :) let me see if I can move this into instance initiation stage instead of function invoking stage 🤔